backend/src/test/java/capstone/backend/controller/OrderToSupplierControllerTest.java

```diff
@@ -0,0 +1,186 @@
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static capstone.backend.mapper.OrderItemMapper.mapOrderItem;
+import static capstone.backend.mapper.OrderToSupplierMapper.mapOrder;
+import static capstone.backend.mapper.SupplierMapper.mapSupplier;
+import static capstone.backend.utils.OrderItemTestUtils.sampleOrderItem;
+import static capstone.backend.utils.OrderToSupplierTestUtils.sampleOrder;
+import static capstone.backend.utils.OrderToSupplierTestUtils.sampleOrderDTO;
+import static capstone.backend.utils.ProductTestUtils.sampleProduct;
+import static capstone.backend.utils.ProductTestUtils.sampleProductWithId;
+import static capstone.backend.utils.SupplierTestUtils.sampleSupplier;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+@Testcontainers
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class OrderToSupplierControllerTest {
+
+    @Autowired
+    OrderToSupplierController controller;
+    @Autowired
+    OrderToSupplierRepo orderRepo;
+    @Autowired
+    ProductRepo productRepo;
+    @Autowired
+    SupplierRepo supplierRepo;
+    @Autowired
+    OrderItemRepo orderItemRepo;
+    @Autowired
+    TestRestTemplate restTemplate;
+    @Autowired
+    ControllerTestUtils utils;
+    String BASEURL = "/api/orders_suppliers";
+
+    @Container
+    public static PostgreSQLContainer container = new PostgreSQLContainer()
+            .withDatabaseName("pos")
+            .withUsername("pos")
+            .withPassword("pos");
+
+    @DynamicPropertySource
+    static void postgresqlProperties(DynamicPropertyRegistry registry) {
+        registry.add("spring.datasource.url", container::getJdbcUrl);
+        registry.add("spring.datasource.password", container::getPassword);
+        registry.add("spring.datasource.username", container::getUsername);
+    }
+
+    @BeforeEach
+    public void clearDB() {
+        orderRepo.deleteAll();
+        orderItemRepo.deleteAll();
+        productRepo.deleteAll();
+        supplierRepo.deleteAll();
+    }
+
+    @Test
+    void containerIsRunning() {
+        assertTrue(container.isRunning());
+    }
+
+
+    @Test
+    void getAllOrders() {
+        //GIVEN
+        Supplier supplier = supplierRepo.save(sampleSupplier());
+        Product product = productRepo.save(sampleProduct().withSuppliers(List.of(supplier)));
+        OrderItem orderItem = orderItemRepo.save(sampleOrderItem().withProduct(product));
+        OrderToSupplier order = orderRepo.save(new OrderToSupplier(1L, List.of(orderItem), supplier));
+        HttpHeaders headers = utils.createHeadersWithJwtAuth();
```

Comment:
seems to be used in every test...move to before each?

Reply:
I need the return data to get the ids of each item I save in the different dbs for the equals methods.
I wish there was a better way